app/livechat/imports/server/rest/sms.js

```diff
@@ -77,30 +86,57 @@ API.v1.addRoute('livechat/sms-incoming/:service', {
 			},
 		};
 
-		sendMessage.message.attachments = sms.media.map((curr) => {
+		let file;
+		const attachments = Promise.await(Promise.all(sms.media.map(async (curr) => {
+			const { url: twilioUrl, contentType } = curr;
+
+			if (!file) {
+				const details = {
+					name: 'Upload File',
+					type: contentType,
+					rid,
+					visitorToken: token,
+				};
+
+				const upload = await getUploadFile(details, twilioUrl);
+				file = Object.assign({}, { _id: upload._id, name: upload.name, type: upload.type });
+			}
+
+			const url = FileUpload.getPath(`${ file._id }/${ encodeURI(file.name) }`);
+
 			const attachment = {
-				message_link: curr.url,
+				message_link: url,
 			};
 
-			const { contentType } = curr;
 			switch (contentType.substr(0, contentType.indexOf('/'))) {
 				case 'image':
-					attachment.image_url = curr.url;
+					attachment.image_url = url;
 					break;
 				case 'video':
-					attachment.video_url = curr.url;
+					attachment.video_url = url;
 					break;
 				case 'audio':
-					attachment.audio_url = curr.url;
+					attachment.audio_url = url;
 					break;
 			}
 
 			return attachment;
-		});
+		})));
+
+		const message = {
+			_id: Random.id(),
+			rid: (room && room._id) || Random.id(),
```

Comment:
```suggestion
			rid,
```
you have created a `rid` variable before at https://github.com/RocketChat/Rocket.Chat/pull/18491/files#diff-e671fbe389e71813da53c895ea42044bR78